backend/model/model.go

```diff
@@ -364,6 +365,12 @@ type SetupEvent struct {
 	Type      MarkBackendSetupType `gorm:"uniqueIndex:idx_project_id_type"`
 }
 
+type ProjectFilterSettings struct {
+	Model
+	ProjectID                  int  `gorm:"uniqueIndex"`
```

Comment:
could remove the `Model` reference and make this could be the primary key to make it clear there is a 1-1 mapping with the `projects` table

Reply:
What's the syntax here? Per the [docs](https://gorm.io/docs/belongs_to.html), I tried changing to:

```go
type ProjectFilterSettings struct {
	Model
	ProjectID                  int
	Project                    Project
	FilterSessionsWithoutError bool `gorm:"default:false"`
}
```

and got this error:

```
2023/05/04 14:38:48 /Users/ericthomas/code/highlight/backend/model/model.go:1334
[error] invalid field found for struct github.com/highlight-run/highlight/backend/model.ProjectFilterSettings's field ProjectID, need to define a valid foreign key for relations or it need to implement the Valuer/Scanner interface
```

Reply:
i think we'd want this to be 
```go
type ProjectFilterSettings struct {
	Project                     *Project
	ProjectID                  int `gorm:"foreignKey:ID;"`
	FilterSessionsWithoutError bool `gorm:"default:false"`
}
```

Reply:
Thanks. I guess we need a pointer here (c22dbd610c60cb5f6427222af3f8eac080de880e). 
I kept `Model` because it's nice to have the created/updated at columns.